src/export/Model/Data/Products/CollateProductData.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+namespace Aligent\FredhopperExport\Model\Data\Products;
+
+use Aligent\FredhopperCommon\Model\Config\GeneralConfig;
+use Magento\Framework\Serialize\Serializer\Json;
+
+class CollateProductData
+{
+    /**
+     * @param Json $json
+     * @param GeneralConfig $generalConfig
+     */
+    public function __construct(
+        private readonly Json $json,
+        private readonly GeneralConfig $generalConfig
+    ) {
+    }
+
+    /**
+     * Collect store records for each product into a single array
+     *
+     * @param array $productData
+     * @return array
+     */
+    public function execute(array $productData): array
+    {
+
+        $defaultStore = $this->generalConfig->getDefaultStore();
+        $productStoreData = [];
+
+        foreach ($productData as $row) {
+            $productStoreData[$row['product_id']] = $productStoreData[$row['product_id']] ?? [];
+            $productStoreData[$row['product_id']]['stores'][$row['store_id']] =
+                $this->json->unserialize($row['attribute_data']);
```

Comment:
Now I'm hunting for it, I assume `$row['attribute_data']` can't ever be `empty()` ?

Reply:
I guess it could be if someone manually updated the database to have an empty string...